plan/preprocess.go

```diff
@@ -283,6 +294,33 @@ func (p *preprocessor) checkDropTableGrammar(stmt *ast.DropTableStmt) {
 	}
 }
 
+func (p *preprocessor) checkNonUniqTableAlias(stmt *ast.Join) {
+	if !p.isParentJoin {
+		p.tableAliasInJoin = append(p.tableAliasInJoin, make(map[string]interface{}))
+	}
+	tableAliases := p.tableAliasInJoin[len(p.tableAliasInJoin)-1]
+	if err := isTableAliasDuplicate(stmt.Left, tableAliases); err != nil {
+		p.err = err
+		return
+	}
+	if err := isTableAliasDuplicate(stmt.Right, tableAliases); err != nil {
+		p.err = err
+		return
+	}
+	p.isParentJoin = true
+}
+
+func isTableAliasDuplicate(node ast.ResultSetNode, tableAliases map[string]interface{}) error {
+	if ts, ok := node.(*ast.TableSource); ok {
+		_, exists := tableAliases[ts.AsName.L]
+		if len(ts.AsName.L) != 0 && exists {
+			return ErrNonUniqTable.GenByArgs(ts.AsName)
+		}
+		tableAliases[ts.AsName.L] = nil
```

Comment:
I think we should only add `ts.AsName.L` into `tableAliases` only when `len(ts.AsName.L) != 0`?

Reply:
It doesn't matter because there's nothing done if `len(ts.AsName.L)  == 0`